### PR TITLE
Update the react-native-windows-init template to use the managed library the default (instead of the shared csproj file).

### PR DESCRIPTION
**Pull request by dannyvv**
###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/4947)

### CODE REVIEW

File: vnext/local-cli/generator-windows/templates/cs/src/App.xaml.cs [hunk 1]
```diff
@@ -3,32 +3,33 @@
 namespace <%=ns%>
 {
     sealed partial class App : ReactApplication
+{
```
Comment (asklar) [inline vnext/local-cli/generator-windows/templates/cs/src/App.xaml.cs:6]: looks like indentation got broken/wrong here?

**Comment by asklar** [comment]
@dannyvv what's the diff between the shared project and the library? (i.e. trying to understand why we were not using this library until now :) )

File: vnext/local-cli/generator-windows/templates/cs/src/App.xaml.cs (hunk 1 above)
Comment (dannyvv) [reply vnext/local-cli/generator-windows/templates/cs/src/App.xaml.cs:6]: yeah, vscode keeps getting the wrong settings for our project and format it wrong. Haven't had time to figure out what setting is causing this...

---
In reply to: [426986471](https://github.com/microsoft/react-native-windows/pull/4947#discussion_r426986471) [](ancestors = 426986471)

**Comment by dannyvv** [comment]
shared project means that all the code from Microsoft.ReactNative.Managed is compiled inside the app as source. Somwhat like adding `<Sources include="..\..\microsoft-reactnative-windows\**\*.cs />` So if you have multiple projects referencing the shared one, there are two problems:
1) You are compiling the same c# files over and over
2) We'll run into type identity issues

This will also be needed for moving to nuget pacakges.

It wasn't the default yet because it was a small breaking change (see change to app.xaml.cs) Since there would be a similar small change for the codegen I opted to wait until the codegen was done, so I could giev advice in an error essage of the codegen with a precise error message. Since this has moved to 0.63 it is okay to just make it the deault now and get some mileage already.


---
In reply to: [630531105](https://github.com/microsoft/react-native-windows/pull/4947#issuecomment-630531105) [](ancestors = 630531105)

File: vnext/local-cli/generator-windows/templates/cs/src/App.xaml.cs (hunk 1 above)
Comment (dannyvv) [reply vnext/local-cli/generator-windows/templates/cs/src/App.xaml.cs:6]: it was the <% in the namespace throwing vscode off....

---
In reply to: [427348372](https://github.com/microsoft/react-native-windows/pull/4947#discussion_r427348372) [](ancestors = 427348372,426986471)

**Comment by NickGerleman** [comment]
FYI @dannyvv please talk to Khalef if you still want this backported. We've stopped auto backporting things and are working out a triage process right now for if we do want to. I'm not sure if the prerequisite changes were backported already though, which would make this one hairy.

**Pull request merged**